### PR TITLE
CLN: follow-ups, whatsnew notes, use tm.get_obj

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/dtypes/common.py [hunk 1]
```diff
@@ -128,10 +128,12 @@ def ensure_python_int(value: int | np.integer) -> int:
...
-    if not is_scalar(value):
-        raise TypeError(
-            f"Value needs to be a scalar value, was type {type(value).__name__}"
-        )
+    if not (is_integer(value) or is_float(value)):
+        if not is_scalar(value):
+            raise TypeError(
+                f"Value needs to be a scalar value, was type {type(value).__name__}"
+            )
+        raise TypeError(f"Wrong type {type(value)} for value {value}")
```
Comment (mroeschke) [inline pandas/core/dtypes/common.py:136]: Is this new error message tested anywhere?

Comment (jbrockmendel) [reply pandas/core/dtypes/common.py:136]: yes, in RangeIndex arithmetic

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -839,15 +839,16 @@ Sparse
...
 - Bug in :meth:`BooleanArray.__eq__` and :meth:`BooleanArray.__ne__` raising ``TypeError`` on comparison with an incompatible type (like a string). This caused :meth:`DataFrame.replace` to sometimes raise a ``TypeError`` if a nullable boolean column was included (:issue:`44499`)
 - Bug in :func:`array` incorrectly raising when passed a ``ndarray`` with ``float16`` dtype (:issue:`44715`)
 - Bug in calling ``np.sqrt`` on :class:`BooleanArray` returning a malformed :class:`FloatingArray` (:issue:`44715`)
 - Bug in :meth:`Series.where` with ``ExtensionDtype`` when ``other`` is a NA scalar incompatible with the series dtype (e.g. ``NaT`` with a numeric dtype) incorrectly casting to a compatible NA value (:issue:`44697`)
-- Fixed bug in :meth:`Series.replace` with ``FloatDtype``, ``string[python]``, or ``string[pyarrow]`` dtype not being preserved when possible (:issue:`33484`)
+- Fixed bug in :meth:`Series.replace` with ``FloatDtype``, ``string[python]``, or ``string[pyarrow]`` dtype not being preserved when possible (:issue:`33484`,:issue:`40732`,:issue:`31644`,:issue:`41215`,:issue:25438`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.4.0.rst:850]: Missing ``` ` ``` in the new issue reference

**Review by mroeschke — APPROVED** [review]
LGTM

**Review by jreback — APPROVED** [review]

**Pull request merged**